ruff.toml

```diff
@@ -0,0 +1,17 @@
+# Enable flake8/pycodestyle (`E`), Pyflakes (`F`), flake8-bandit (`S`), and isort (`I`) rules.
+select = ["E", "F", "S", "I"]
+fixable = ["I"]
+
+# Never enforce `E402`, `E501` (line length violations),
+# and `S311` (random number generators)
+ignore = ["E402", "E501", "S311"]
```

Comment:
In principle, yes. Inclusion of this error throws flags on a few files (pix2pix.py, super_res_net.py) that have  `# ignore_header_test` flags at the top of the file. We can individually ignore this rule for those files but it seems sort of hacky. @NickGeneva ?

Reply:
I know theres an import mid file for distributed comms too that would error with 402... case by case ignore is fine when its just a few outliers.

Wouldn't any try except statements trigger this or no?
E.g. given that we have so many opt deps, if we want to include more meaningful error messages for people.

If it does for this, then I would suggest ignoring the issue entirely

Reply:
So I lifted this ignore and put in a few file `ruff: noqa: E402` for the handful of places that had issues. There are a number of `test/` files that follow this pattern:
```python
script_path = os.path.abspath(__file__)
sys.path.append(os.path.join(os.path.dirname(script_path), ".."))

import common
```

Any thoughts?

Reply:
My initial impression is that those tests can just be one directory up and have all the utilities for the tests inside model specific directory. If that's not acceptable, then the imports should use relative imports instead of changing the python path: https://docs.python.org/2/whatsnew/2.5.html#pep-328-absolute-and-relative-imports

Here's a test with relative imports:
```zsh
❯ tree
.
├── common
│   ├── __init__.py
│   └── utils.py
├── __init__.py
└── test_sfno
    ├── __init__.py
    └── test_sfno.py
```

Contents of `test_sfno.py`
```python
from ..common import utils


def test_value():
    print(f"value in module common = {utils.value}")
    assert utils.value == 1
```

pytest output:
```zsh
❯ pytest
==================================================================================================================================================== test session starts ====================================================================================================================================================
platform linux -- Python 3.10.12, pytest-7.4.2, pluggy-1.3.0
rootdir: /home/akshays/Codes/legate/import_test
plugins: anyio-3.7.1
collected 1 item                                                                                                                                                                                                                                                                                                            

test_sfno/test_sfno.py .                                                                                                                                                                                                                                                                                              [100%]

===================================================================================================================================================== 1 passed in 0.01s =====================================================================================================================================================
```

Reply:
Turns out there is not a clean way to do this for now since the @akshaysubr 's above solution introduces an `E999` flag with the linter. I tried out another solution to no avail. For now, I will keep the present solution.